private/manageFishData.py

```diff
@@ -84,76 +106,103 @@ def _make_static_localized_field(fld_name, value):
+            ('territory_id', spot.get_raw('TerritoryType'))]))
+      for spot in XIV.get_sheet('FishingSpot')])
+  
+  SPEARFISHING_NODES = dict([
+      (x['GatheringPointBase'].key,
+      dict([('_id', x['GatheringPointBase'].key),
+            *_decode_spearfishing_node_name(x),
+            ('territory_id', x.get_raw('TerritoryType'))]))
+      for x in XIV.get_sheet('GatheringPoint')
+      if x['GatheringPointBase']['GatheringType'].key == 4])
+  
+  REGIONS = dict([
+      (territory.region_place_name.key,
+       dict(_make_localized_field('name', territory.region_place_name, 'Name')))
+      for territory in TERRITORIES])
+  
+  ZONES = dict([
+      (territory.place_name.key,
+       dict(_make_localized_field('name', territory.place_name, 'Name')))
+      for territory in TERRITORIES])
+
+  ICON_MAP = {
+      '': [
+          (9, 'DEFAULT.png'),
+      ],
+      'action': [
+          (1115, 'powerful_hookset.png'),  # Action[Name="Powerful Hookset"]
+          (1116, 'precision_hookset.png'),  # Action[Name="Precision Hookset"]
+          (60671, 'small_gig.png'),
+          (60672, 'normal_gig.png'),
+          (60673, 'large_gig.png'),
+      ],
+      'status': [
+          (11101, 'intuition.png'),  # Status[Name="Fisher's Intuition"]
+          (11102, 'snagging.png'),  # Status[Name="Snagging"]
+          (11103, 'fish_eyes.png'),  # Status[Name="Fish Eyes"]
+      ]
+  }
+  
+  # Store the dictionaries sorted by key
+  # This makes the generated JS a bit more consistent.
+  WEATHER_RATES = OrderedDict(sorted(WEATHER_RATES.items(), key=lambda t: t[0]))
+  WEATHER_TYPES = OrderedDict(sorted(WEATHER_TYPES.items(), key=lambda t: t[0]))
+  FISHING_NODES = OrderedDict(sorted(FISHING_NODES.items(), key=lambda t: t[0]))
+  SPEARFISHING_NODES = OrderedDict(sorted(SPEARFISHING_NODES.items(), key=lambda t: t[0]))
+  REGIONS = OrderedDict(sorted(REGIONS.items(), key=lambda t: t[0]))
+  ZONES = OrderedDict(sorted(ZONES.items(), key=lambda t: t[0]))
+  
+  KeyValuePair = namedtuple('KeyValuePair', ['key', 'value'])
 
-WEATHER_RATES = dict([
-    (territory.key,
-     dict({'map_id': territory.map.key,
-           'zone_id': territory.place_name.key,
-           'region_id': territory.region_place_name.key,
-           'weather_rates': _collect_weather_rates(territory.weather_rate)}))
-    for territory in TERRITORIES])
-
-WEATHER_TYPES = dict([
-    (weather.key,
-     dict([*_make_localized_field('name', weather, 'Name'),
-           ('icon', '%06u' % weather.get_raw('Icon'))]))
-    for weather in
-    set(reduce(add, [t.weather_rate.possible_weathers for t in TERRITORIES], []))
-    if weather.key != 0])
-
-FISHING_NODES = dict([
-    (spot.key,
-     dict([('_id', spot.key),
-           *_make_localized_field('name', spot['PlaceName'], 'Name'),
-           ('territory_id', spot.get_raw('TerritoryType'))]))
-    for spot in XIV.get_sheet('FishingSpot')])
 
 def _decode_spearfishing_node_name(x):
```

Comment:
I think the `_decode_spearfishing_node_name` function has to be defined before line 152, `SPEARFISHING_NODES`.